app/src/main/java/protect/card_locker/LoyaltyCardListDisplayOptionsManager.java

```diff
@@ -0,0 +1,177 @@
+package protect.card_locker;
+
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.SharedPreferences;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+
+import com.google.android.material.dialog.MaterialAlertDialogBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class LoyaltyCardListDisplayOptionsManager {
+    public final Context mContext;
+
+    private final Runnable mRefreshCardsCallback;
+    private final Runnable mSwapCursorCallback;
+
+    protected SharedPreferences mCardDetailsPref;
+
+    private boolean mShowNameBelowThumbnail;
+    private boolean mShowNote;
+    private boolean mShowBalance;
+    private boolean mShowValidity;
+    private boolean mShowArchivedCards;
+
+    public static class LoyaltyCardDisplayOption {
```

Comment:
I would usually put nested classes at the top, before any fields. But either works.